### DISCUSSION TITLE
Error 400 when fetching address

### DISCUSSION THREAD

**Discussion by evdmeer** [spatie/bpost-address-webservice#8] [Q&A]
I am trying to use the library using the supplied example code, but I am getting an error 400 from Guzzle when I call the validate() function

Exception code: 400
Exception message: Client error: `POST https://webservices-pub.bpost.be/ws/ExternalMailingAddressProofingCSREST_v1/address/validateAddresses` resulted in a `400 Bad Request` response:
{"exceptionId":"7513584400550612864","exceptionCode":"2111","exceptionDescription":[{"value":"Cannot deserialize instanc (truncated...)

At first I thought this was caused by an old guzzle version, but I checked the guzzle version composer fetched, and it is 7.8.1 , which seems to be the latest version.

**Comment by freekmurze** [comment]
We don't use this code actively anymore. 

If you find a fix, we'd appreciate a PR.

**Comment by evdmeer** [comment]
I did find out what the underlying issue is.
If you make a request and do not supply any options, the request is made with an empty ValidateAddressOptions element, which the API does not allow.
If you supply at least 1 option (for example with the withOptions() function ), the request will succeed.
The solution is to not add the ValidateAddressOptions element to the request if no options are provided.
